components/match2/commons/match_group_display.lua

```diff
@@ -102,6 +102,42 @@ function MatchGroupDisplay.MatchGroupById(args)
 	})
 end
 
+--[[
+Displays a singleMatch specified by a bracket ID and matchID.
+]]
+function MatchGroupDisplay.MatchByMatchId(args)
+	local bracketId = args.id or args[1]
+	args.id = bracketId
+	args[1] = bracketId
+	assert(bracketId, 'Missing bracket ID')
+
+	local matches = MatchGroupUtil.fetchMatches(bracketId)
+	assert(#matches > 0, 'No data found for bracketId=' .. bracketId)
+
+	if (#matches > 1) then
+		assert(args.matchid, 'Missing match ID')
+		local matchIdLength = string.len(args.matchid)
+		args.matchid = string.rep('0', 4 - matchIdLength) .. args.matchid
+
+		matches = Table.filter(matches, function(match)
+			local _, matchid = MatchGroupUtil.splitMatchId(match.matchId)
+			return matchid == args.matchid end
+		)
+
+		assert(#matches == 1, 'Match matchId=' .. args.matchid .. ' not found')
```

Comment:
If not using `MatchGroupUtil.fetchMatchForBracketDisplay` (suggested below) then this should be 

```
	local matchId = ...

...

	local matchGroup = MatchGroupUtil.fetchMatchGroup(bracketId)

	local fullMatchId = bracketId .. '_' .. matchId
	local match = matchGroup.matchesById[fullMatchId]

	assert(match, 'Match bracketId= ' .. bracketId .. ' matchId=' .. matchId .. ' not found')
```

Reply:
Implemented as suggested